recipes/imgui/all/conandata.yml

```diff
@@ -29,9 +29,9 @@ sources:
   "1.83":
     url: "https://github.com/ocornut/imgui/archive/v1.83.tar.gz"
     sha256: "ccf3e54b8d1fa30dd35682fc4f50f5d2fe340b8e29e08de71287d0452d8cc3ff"
-  "1.84.1":
-    url: "https://github.com/ocornut/imgui/archive/v1.84.1.tar.gz"
-    sha256: "292ab54cfc328c80d63a3315a242a4785d7c1cf7689fbb3d70da39b34db071ea"
+  # "1.84.1":
+  #   url: "https://github.com/ocornut/imgui/archive/v1.84.1.tar.gz"
+  #   sha256: "292ab54cfc328c80d63a3315a242a4785d7c1cf7689fbb3d70da39b34db071ea"
```

Comment:
no need to comment, git is our friend.

Reply:
I prefer to comment. One could easily write a tool that lets you build versions that are commented. 

If you have to search the git history first that's much more complicated

Reply:
Versions which are commented are not tested anymore and could not work with the current conanfile.py. IMMO, either they are removed from conanfile.yml (as well as patches), either they are kept so that we ensure that conanfile.py and patches are still compliant.

Reply:
The thing is that we mostly remove versions to save resources for Conan Center, which we have to do because those resources are limited, but that does not mean that we have to take away other advantages of having once published versions permanent in the `conandata.yml`.

1) Keeping old versions commented makes it clearer to understand how the Conan Center remote got populated. Having versions in the remote but not in git can lead to confusion (e.g. https://github.com/conan-io/conan-center-index/issues/9452) and I think that will increase with Conan Center growing and the number of versions we remove growing as well. Confusion aside, even if people are aware of that it's still quicker to understand the history of the Conan Center remote if it is commented. (We had a similar discussion about obsolete packages and I think the consensus there was to keep it in git as well to prevent things like accidentally re-use of package names and for transparency)

2) Sometimes there is the need to build older versions for whatever reason. Once we comment them, we don't test them anymore and they might start to fail at some points - that's true. However, since Conan can't build them without manual changes in the `conandata.yml` file or without using some tool for that, it should be clear enough that there isn't official community support for it. 

I just don't see the point of making things harder for a very little advantage, which seems to be only to have a smaller `conandata.yml` file and I don't think that the file size there will be an actual problem anytime soon.